### ISSUE TITLE
allow longer challenge names for ctfs with short names

### ISSUE THREAD

**Issue by Grazfather** [OpenToAllCTF/OTA-Challenge-Bot#148]
We basically say 10 chars for ctf name and 10 for chal + 1 for the underscore to hit the 21 char channel name limit.

For short ctf names let's loosen up the restriction on the number of characters allowed for the challenge name.

**Comment by Grazfather** [comment]
```diff
diff --git a/handlers/challenge_handler.py b/handlers/challenge_handler.py
index ba8097d..40db05e 100644
--- a/handlers/challenge_handler.py
+++ b/handlers/challenge_handler.py
@@ -179,7 +179,7 @@ class AddChallengeCommand(Command):
         if not ctf:
             raise InvalidCommand("Add challenge failed: You are not in a CTF channel.")

-        if len(name) > 10:
+        if len(name) > (20 - len(ctf.name):
             raise InvalidCommand("Add challenge failed: Challenge name must be <= 10 characters.")

         # Check for invalid characters
```

should do the trick, no time to test rn.

**Comment by Kileak** [comment]
For this, `renamechallenge` and `renamectf` also need to be reworked. 

We need to do a pre-check on all challenges, if renaming the ctf would break the length.

**Comment by Kileak** [comment]
Fixed by 26714482f2aa9da98f329686f5a56dd8e58d5636
